src/pybind/mgr/affinity/module.py

```diff
@@ -0,0 +1,152 @@
+
+"""
+affinity module is aimed at managing multiple availability zone situations
+that need local read affinity.
+local read affinity in ceph means that all primary OSDs of a pool are from the same az.
+
+See doc/mgr/affinity.rst for more info.
+
+currently the only implemented feature is the ability to create such pool, by creating the correct CRUSH rule.
+in the futuer we will implement more features to monitor and manage failures, etc.
+"""
+
+from mgr_module import MgrModule, CommandResult, HandleCommandResult
+from threading import Event
+
+import os
+import json
+import subprocess
+from tempfile import mkstemp
+from mako.template import Template
```

Comment:
@alfredodeza ?

Reply:
Not sure what I can add here... are you wondering about the dependency? I couldn't find `mako` being defined anywhere in the Ceph source tree so this would imply that this import is a no-go.